src/components/student/student.repository.ts

```diff
@@ -0,0 +1,260 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { StudentEntity } from './entity/student.entity';
+import { Batch, Repository, SelectQueryBuilder } from 'typeorm';
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { DepartmentEntity } from '../department/entity/department.entity';
+import { BatchEntity } from '../batch/entity/batch.year.entity';
+import { BatchDetailsEntity } from '../batch/entity/batch.details.entity';
+import { AttendanceEntity } from '../attendance/entity/attendance.entity';
+import { format, parse } from 'date-fns';
+
+@Injectable()
+export class StudentRepository {
+  constructor(
+    @InjectRepository(StudentEntity)
+    private studentRep: Repository<StudentEntity>,
+
+    @InjectRepository(DepartmentEntity)
+    private departmentRepo: Repository<DepartmentEntity>,
+
+    @InjectRepository(BatchEntity)
+    private batchRepo: Repository<BatchEntity>,
+
+    @InjectRepository(AttendanceEntity)
+    private attendanceRepo: Repository<AttendanceEntity>,
+  ) {}
+
+  public async getAllStudents(): Promise<StudentEntity[]> {
+    return await this.studentRep.find();
+  }
+
+  public async getStudentById(id: string): Promise<StudentEntity> {
+    const studentEntity=  await this.studentRep.findOne({ where: { id: id } });
```

Comment:
You are checking existence and throwing error where you used this method: [here](https://github.com/kevit-siddh-kothari/nest-college-erp/pull/2/files#diff-134bbbdabd386ced6e43d79dcd3cbe7ade01c547ee86e39e09c15a54b31d7279R15). you can skip if condition.

Reply:
OK